src/AutoRest.CSharp/MgmtTest/Extensions/CodeWriterExtensions.cs

```diff
@@ -104,12 +105,50 @@ private static CodeWriter AppendFrameworkTypeValue(this CodeWriter writer, CShar
             if (type.FrameworkType == typeof(BinaryData))
                 return writer.AppendBinaryData(exampleValue);
 
+            if (type.FrameworkType == typeof(DataFactoryElement<>))
+                return writer.AppendDataFactoryElementValue(type, exampleValue);
+
             if (exampleValue.Schema is ObjectSchema objectSchema)
                 return writer.AppendComplexFrameworkTypeValue(objectSchema, type.FrameworkType, exampleValue);
 
             return writer.AppendRawValue(exampleValue.RawValue, type.FrameworkType, exampleValue.Schema.Type);
         }
 
+        private static CodeWriter AppendDataFactoryElementValue(this CodeWriter writer, CSharpType type, ExampleValue exampleValue)
+        {
+            if (type.FrameworkType != typeof(DataFactoryElement<>))
+                throw new ArgumentException("DataFactoryElement<> is expected but got: " + type.ToString());
```

Comment:
We kind of do not need this because we kind of ensures this in our implementation.
How about we replace this guard by `Debug.Assert` which will be automatically removed in the release version of the generator.

Reply:
it's better to throw error as soon as possible to help on troubleshooting